### PR TITLE
newdevice: Added extrahop detection

### PR DESCRIPTION
**Pull request by geordish**
Fix for #6080 

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/discovery/os/linux.inc.php [hunk 1]
```diff
@@ -12,6 +12,8 @@
         } elseif (snmp_get($device, 'fwVersion.1', '-Osqnv', 'UBNT-AirFIBER-MIB', 'ubnt') !== false) {
             $os = 'airos-af';
         }
+    } elseif (snmp_get($device, 'EXTRAHOP-MIB::extrahopInfoVersionString', '-Osqnv') !== false) {
```
Comment (laf) [inline includes/discovery/os/linux.inc.php]: I know the other gets are like this but can you shift the EXTRAHOP-MIB to the end of the function:

`snmp_get($device, 'extrahopInfoVersionString', '-Osqnv', 'EXTRAHOP-MIB')`

File: includes/polling/os/extrahop.inc.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright (c) 2017 Dave Bell <me@geordish.org>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+$version = snmp_get($device, 'EXTRAHOP-MIB::extrahopInfoVersionString', '-Oqv');
```
Comment (laf) [inline includes/polling/os/extrahop.inc.php]: `snmp_get($device, 'extrahopInfoVersionString', '-Oqv', 'EXTRAHOP-MIB')`

**Comment by geordish** [comment]
Don't think I can easily move the MIB, or the discovery doesn't work :(

**Comment by laf** [comment]
For the MIB dir, you just need to specify the dir to look in for OS detection using the extra param in snmp_get() (any other detection will be fine as it will know the OS and therefore look in the correct dir).

**Pull request merged**